File: src/main/java/org/example/SessionManager.java
package org.example;

import lombok.Data;
import lombok.Getter;
import org.example.DTO.Car;
import org.example.DTO.User;


public enum SessionManager {
    INSTANCE;

    private User user;
    private Car car;

    public Car getCar() {
        return car;
    }

    public void setCar(Car car) {
        this.car = car;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

}

File: lib/utils/session_manager.dart
import '../models/user.dart';
import '../models/car.dart';

class SessionManager {
  static final SessionManager instance = SessionManager._internal();

  factory SessionManager() {
    return instance;
  }

  SessionManager._internal();

  User? _user;
  Car? _car;

  User? getUser() {
    return _user;
  }

  void setUser(User? user) {
    _user = user;
  }

  Car? getCar() {
    return _car;
  }

  void setCar(Car? car) {
    _car = car;
  }

  void clear() {
    _user = null;
    _car = null;
  }

  bool isLoggedIn() {
    return _user != null;
  }
}
